### PR TITLE
Add more unit tests for ZoneGetter change in NEG controller

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula 
cc @bowei

### CODE REVIEW

File: pkg/utils/zonegetter/fake.go [hunk 1]
```diff
@@ -77,6 +77,49 @@ func AddFakeNodes(zoneGetter *ZoneGetter, newZone string, instances ...string) e
 	return nil
 }
 
+// AddFakeNodeWithNoProviderID adds fake node without providerID to the ZoneGetter.
+func AddFakeNodeWithNoProviderID(zoneGetter *ZoneGetter, instance string) error {
```
Comment (swetharepakula) [inline pkg/utils/zonegetter/fake.go:81]: how about a generic AddFakeNode which takes in a node object as a parameter? 

or do we think these 2 functions are the last 2 types of nodes we need?

Comment (songrx1997) [reply pkg/utils/zonegetter/fake.go]: Done

File: pkg/utils/zonegetter/fake.go [hunk 2]
```diff
@@ -77,6 +77,37 @@ func AddFakeNodes(zoneGetter *ZoneGetter, newZone string, instances ...string) e
...
+// AddFakeNode adds fake node to the ZoneGetter.
+func AddFakeNode(zoneGetter *ZoneGetter, node *apiv1.Node) error {
+	if err := zoneGetter.nodeInformer.GetIndexer().Add(node); err != nil {
+		return err
+	}
+	return nil
+}
+
+// AddFakeNodeWithInvalidProviderID adds fake node with invalid providerID to the ZoneGetter.
+func AddFakeNodeWithInvalidProviderID(zoneGetter *ZoneGetter, instance string) error {
```
Comment (swetharepakula) [inline pkg/utils/zonegetter/fake.go:89]: do we still need this function with the AddFakeNode?

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**